libs/community/tests/integration_tests/vectorstores/test_aerospike.py

```diff
@@ -0,0 +1,838 @@
+            dimensions=10,
+            vector_distance_metric=types.VectorDistanceMetric.DOT_PRODUCT,
+        )
+        aerospike.add_texts(
+            ["foo", "bar", "baz"],
+            ids=["1", "2", "3"],
+            index_name=index_name,
+            set_name=set_name,
+        )  # blocking
+
+        """
+        foo vector = [1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 0.0]
+        far vector = [1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 3.0]
+        dot product = 9.0
+        dot product distance = dot product * -1 = -9.0
+        """
+        expected = -9.0
+        output = aerospike.similarity_search_with_score(
+            "far", index_name=index_name, k=3
+        )
+
+        _, actual_score = output[2]
+
+        assert actual_score == expected
+
+    def test_euclidean_distance(self, aerospike: Aerospike, admin_client: Any) -> None:
+        """Test dot product distance."""
+        from aerospike_vector_search import types
+
+        index_name = set_name = get_func_name()
+        admin_client.index_create(
+            namespace=TEST_NAMESPACE,
+            sets=set_name,
+            name=index_name,
+            vector_field=VECTOR_KEY,
+            dimensions=10,
+            vector_distance_metric=types.VectorDistanceMetric.SQUARED_EUCLIDEAN,
+        )
+        aerospike.add_texts(
+            ["foo", "bar", "baz"],
+            ids=["1", "2", "3"],
+            index_name=index_name,
+            set_name=set_name,
+        )  # blocking
+
+        """
+        foo vector = [1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 0.0]
+        far vector = [1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 3.0]
+        euclidean distance = 9.0
+        """
+        expected = 9.0
+        output = aerospike.similarity_search_with_score(
+            "far", index_name=index_name, k=3
+        )
+
+        _, actual_score = output[2]
+
+        assert actual_score == expected
+
+    def test_as_retriever(self, aerospike: Aerospike, admin_client: Any) -> None:
+        index_name = set_name = get_func_name()
+        admin_client.index_create(
+            namespace=TEST_NAMESPACE,
+            sets=set_name,
+            name=index_name,
+            vector_field=VECTOR_KEY,
+            dimensions=10,
+        )
+        aerospike.add_texts(
+            ["foo", "foo", "foo", "foo", "bar"],
+            ids=["1", "2", "3", "4", "5"],
+            index_name=index_name,
+            set_name=set_name,
+        )  # blocking
+
+        aerospike._index_name = index_name
+        retriever = aerospike.as_retriever(
+            search_type="similarity", search_kwargs={"k": 3}
+        )
+        results = retriever.invoke("foo")
+        assert len(results) == 3
+        assert all([d.page_content == "foo" for d in results])
+
+    def test_as_retriever_distance_threshold(
+        self, aerospike: Aerospike, admin_client: Any
+    ) -> None:
+        from aerospike_vector_search import types
```

Comment:
**Issue**: The test method 'test_as_retriever_distance_threshold' uses a hardcoded 'COSINE' metric which might limit the flexibility of the test. Consider parameterizing the test to cover more scenarios. <br> **Fix**: Parameterize the test to allow different distance strategies to be tested dynamically. <br> **Code Suggestion**: 
 ```
 -        aerospike._distance_strategy = DistanceStrategy.COSINE
 +        aerospike._distance_strategy = pytest.param(DistanceStrategy.COSINE, DistanceStrategy.EUCLIDEAN, ...)
 ```